### PR TITLE
Postgresql tests via plugin

### PR DESCRIPTION
**Pull request by fschulze**

### CODE REVIEW

File: postgresql/tox.ini [hunk 1]
```diff
@@ -14,6 +14,14 @@ deps =
     pytest-catchlog
     pytest-flakes
     pytest-pep8
+    -e
+    {toxinidir}/../common
+    -e
+    {toxinidir}/../server
```
Comment (hpk42) [inline postgresql/tox.ini]: putting "-e" and the location in different lines is intentional?   I thought it needs to be on the same line.

File: server/devpi_server_pytest/__init__.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+def pytest_addoption(parser):
+    parser.addoption(
+        "--slow", action="store_true", default=False,
+        help="run slow tests involving remote services (pypi.python.org)")
+    parser.addoption(
+        "--backend", action="store",
+        help="run tests with specified dotted name backend")
```
Comment (hpk42) [inline server/devpi_server_pytest/__init__.py]: hum, this introduces a "completion" conflict when i type "vim devpi_serve<TAB>" which currently does not exist.   You could also put the plugin into the test_devpi_server directory  which kind of makes sense i'd think.

File: postgresql/tox.ini (hunk 1 above)
Comment (fschulze) [inline postgresql/tox.ini]: Yes, it seems this is a bug in tox, it doesn't work when it's on one line.

File: server/devpi_server_pytest/__init__.py (hunk 2 above)
Comment (fschulze) [inline server/devpi_server_pytest/__init__.py]: I've moved the plugin. I had problems before, which I thought where caused by placing it in test_devpi_server, but they were unrelated.

**Comment by fschulze** [comment]
Moving the plugin into test_devpi_server doesn't work after all: https://travis-ci.org/fschulze/devpi/jobs/230385547

**Comment by fschulze** [comment]
Is there a way to fix the issue? Or should I move the plugin to ``pytest_devpi_server`` to avoid the auto complete conflict?

File: postgresql/tox.ini (hunk 1 above)
Comment (obestwalter) [reply postgresql/tox.ini]: Not a bug - a documented limitation: "Each line specifies a file, a URL or a package name." [*](http://tox.readthedocs.io/en/latest/config.html#confval-deps=MULTI-LINE-LIST) (so, no options) - there is no way to add options as each line is taken as is and not split before passed to pip. If you need options you either need to use a requirements file or stick you pip invocation directly into commands.

But you worked around this limitation beautifully :)

Comment (fschulze) [reply postgresql/tox.ini]: Options could be supported by doing (pseudocode):
```python
if option.startswith('-'):
    parts = option.split(None, 1)
    args.extend(parts)
```
or am I missing something?

Comment (obestwalter) [reply postgresql/tox.ini]: I agree that it seems fixable with a bit of extra effort. Question is, if someone wants it strong enough to add it. I never missed it and was not even aware of that limitation until I saw this :)

Comment (fschulze) [reply postgresql/tox.ini]: True

Comment (obestwalter) [reply postgresql/tox.ini]: btw sorry for necromancing this - I just realized that the last comment here was months ago :D

**Pull request merged**